local-compose.yml

```diff
@@ -0,0 +1,33 @@
+nginx:
+    extends:
+        file: docker-compose.yml
+        service: nginx
+    build: .
+    mem_limit: 128g
+    links:
+        - consul:consul
+    environment:
+        - CONSUL=consul
```

Comment:
You have `link` repeated in this service, and if you have a link for `- consul:consul` then you'll automatically have the hostname `consul` in your `/etc/hosts` in the container (or in the Docker-hijacked DNS, if I understand correctly).